tutorials/integration.md

```diff
@@ -534,27 +595,37 @@ const gsnPaymasterRejection = async () => {
 
 ```
 
-
+&nbsp;
 
 This function listens for events, which is free, so we can use the provider rather than 
 a signer. It is also standard `ethers.js`, and provided here for completeness.
 ```javascript
-const listenToEvents = async () => {
+onst listenToEvents = async () => {
+	await startGsn()
+
 	const contract = await new ethers.Contract(
-		flagAddr, flagAbi, provider);
+		conf.ourContract, data.abi, provider);
+
+        window.app.listenContract = contract
 ```
 
-This is an easy way to listen to events in `ethers.js`, and to parse them
-once they are received
+&nbsp;
+
+This is an easy way to listen to events in `ethers.js` version 5. When you
+listen for an event with n parameters, those values are the first n 
+parameters. The final parameter is the complete event information.
```

Comment:
This information is very useful but is not a documentation of GSN in my understanding of things.

Reply:
I wrote whatever I wished I had known when I started the project, to make things easier for implementers, because this is a tutorial, not a reference. If you insist I'll delete this and all information like it, but I think that would be a mistake.

Reply:
This means your target audience is basically yourself from the past. I don't think this is a right assumption.

For those who already know the 'extra stuff' in your tutorial it will be hard to filter out the water.

For those who don't know it the given detail will be insufficient.

Large portions of this text do not relate to GSN and this is a missed opportunity to describe the system to those who are ready to learn about it in isolation from basic programming techniques, JavaScript syntax and specifics of libraries used in the tutorial.